libraries/common/stream.effekt

```diff
@@ -30,6 +30,17 @@ effect stop(): Nothing
 def for[A] { stream: () => Unit / emit[A] } { action: A => Unit }: Unit =
   returning::for[A, Unit]{stream}{action}
 
+/// Runs the stream as a message queue,
+/// handling each of its elements that can produce new elements ("messages"),
+/// until a fixed point is reached.
+def fix[T] { one: T => Unit / emit[T] } { stream: => Unit / emit[T] }: Unit =
+  try {
+    stream()
+  } with emit[T] { value =>
+    fix[T] {one} { one(value) }
+    resume(())
+  }
```

Comment:
Does this do the correct thing wrt stack? (i.e. not building up extra unnecessary frames?)

Reply:
We could use the VMTests to actually measure and test this :)

Reply:
I added a VM test, but I cannot really tell from the output if it does the correct thing 🤔